src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -265,6 +265,52 @@ public void CustomFunction()
             Assert.Equal(6.0, result.ToObject());
         }
 
+        [Fact]
+        public void ConvertToDisplayNames()
+        {
+            var engine = new RecalcEngine();
+            RecordType r1 = new RecordType()
+                .Add(new NamedFormulaType("Num", FormulaType.Number))
+                .Add(new NamedFormulaType("B", FormulaType.Boolean))
+                .WithDisplayNames(new Dictionary<string, string>() { {"Num", "DisplayNum" }, { "B", "DisplayB" } });
+
+            var displayExpressions = engine.GetDisplayExpression("If(B, Num, 1234)", r1);
+
+            Assert.Equal("If(DisplayB, DisplayNum, 1234)", displayExpressions);
+        }
+
+        
+
+        [Fact]
+        public void ConvertToDisplayNameIsNoOp()
```

Comment:
some other tests:

- can we have a mixed expression? (display + logical names) 

- calling GetDisplayExpression when WithDisplayNames() was never called?

- collision case (that throws the new exception)